google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -490,6 +516,13 @@ public TransactionContext resetForRetry() {
             return new SessionPoolTransactionContext(delegate.resetForRetry());
           }
         } catch (SessionNotFoundException e) {
+          sessionReplacementCount++;
+          if (sessionReplacementCount > MAX_SESSION_REPLACES) {
```

Comment:
same as above.. just leave this comment open until there is a better strategy defined for prev instance